cpp/src/c_api/eigenvector_centrality.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2021-2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cugraph_c/algorithms.h>
+
+#include <c_api/abstract_functor.hpp>
+#include <c_api/centrality_result.hpp>
+#include <c_api/graph.hpp>
+#include <c_api/resource_handle.hpp>
+#include <c_api/utils.hpp>
+
+#include <cugraph/algorithms.hpp>
+#include <cugraph/detail/utility_wrappers.hpp>
+#include <cugraph/graph_functions.hpp>
+
+#include <optional>
+
+namespace {
+
+struct eigenvector_centrality_functor : public cugraph::c_api::abstract_functor {
+  raft::handle_t const& handle_;
+  cugraph::c_api::cugraph_graph_t* graph_;
+  double epsilon_;
+  size_t max_iterations_;
+  bool do_expensive_check_;
+  cugraph::c_api::cugraph_centrality_result_t* result_{};
+
+  eigenvector_centrality_functor(cugraph_resource_handle_t const* handle,
+                                 cugraph_graph_t* graph,
+                                 double epsilon,
+                                 size_t max_iterations,
+                                 bool do_expensive_check)
+    : abstract_functor(),
+      handle_(*reinterpret_cast<cugraph::c_api::cugraph_resource_handle_t const*>(handle)->handle_),
+      graph_(reinterpret_cast<cugraph::c_api::cugraph_graph_t*>(graph)),
+      epsilon_(epsilon),
+      max_iterations_(max_iterations),
+      do_expensive_check_(do_expensive_check)
+  {
+  }
+
+  template <typename vertex_t,
+            typename edge_t,
+            typename weight_t,
+            bool store_transposed,
+            bool multi_gpu>
+  void operator()()
+  {
+    // FIXME: Think about how to handle SG vice MG
```

Comment:
Should we do anything different? (compared to PageRank & Katz centrality?)

Reply:
Regarding the FIXME?

I should probably create an issue and delete all of the FIXMEs that are propagating here.  The issue is to come up with a way to have an SG only version of cugraph.  Some customers don't need an MG version and an SG only package of cugraph would be lighter weight for them to install.  But this aspect of the C API implementation would need to be addressed somehow.

However it is addressed, it would be addressed the same in each API.